pkg/client/typed/dynamic/client.go

```diff
@@ -79,28 +79,39 @@ func NewClient(conf *restclient.Config) (*Client, error) {
 	return &Client{cl: cl}, nil
 }
 
-// Resource returns an API interface to the specified resource for
-// this client's group and version. If resource is not a namespaced
-// resource, then namespace is ignored.
-func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *ResourceClient {
-	return &ResourceClient{
+// Resource returns an API interface to the specified resource for this client's
+// group and version. If resource is not a namespaced resource, then namespace
+// is ignored. parameterCodec sets parameter codec used in the List, Watch,
+// and DeleteCollection methods. defaultParameterEncoder will be used if no
+// parameterCodec is provided.
+func (c *Client) Resource(resource *unversioned.APIResource, namespace string, parameterCodec ...runtime.ParameterCodec) *ResourceClient {
```

Comment:
@krousey @lavalamp how about making parameterCodec a variadic argument to Resource()?